packages/eslint-plugin/docs/rules/consistent-type-imports.md

```diff
@@ -88,11 +88,70 @@ type T = import('Foo').Foo;
 const x: import('Bar') = 1;
 ```
 
-## Usage with `emitDecoratorMetadata`
+## Caveat: `@decorators` + `experimentalDecorators: true` + `emitDecoratorMetadata: true`
```

Comment:
[Refactor] This is a pretty huge caveat that takes up multiple windows-worth of content for my desktop monitors. I'd expect that most users probably won't read it... even though the information is good & pertinent.

What do you think about moving roughly everything after the _TL;DR_ and before the next heading into a linked blog post? As in, have this rule's docs contain the high-level important conclusions, and any explanations >=1 paragraph long moved into a deeper read format?

I'm in favor to the point of requesting changes.